kie-server-parent/kie-server-tests/kie-server-integ-tests-controller/src/test/java/org/kie/server/integrationtests/controller/RestKieControllerManagementIntegrationTest.java

```diff
@@ -0,0 +1,128 @@
+/*
+ * Copyright 2015 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * 
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+*/
+
+package org.kie.server.integrationtests.controller;
+
+import java.util.HashMap;
+
+import org.junit.Test;
+import org.kie.server.api.model.KieContainerResourceList;
+import org.kie.server.api.model.KieContainerStatus;
+import org.kie.server.api.model.ServiceResponse;
+import org.kie.server.controller.api.model.spec.ContainerSpec;
+import org.kie.server.controller.api.model.spec.ServerTemplate;
+import org.kie.server.controller.management.client.exception.KieServerControllerHTTPClientException;
+import org.kie.server.integrationtests.shared.KieServerAssert;
+
+import static org.junit.Assert.*;
+
+public class RestKieControllerManagementIntegrationTest extends KieControllerManagementIntegrationTest<KieServerControllerHTTPClientException> {
+
+    @Override
+    protected void assertNotFoundException(KieServerControllerHTTPClientException e) {
+        assertEquals(404,
+                     e.getResponseCode());
+        assertNotNull(e.getMessage());
+    }
+
+    @Override
+    protected void assertBadRequestException(KieServerControllerHTTPClientException e) {
+        assertEquals(400, e.getResponseCode());
+        assertNotNull(e.getMessage());
+    }
+
+    @Test
+    public void testCreateDuplicateServerTemplate() {
```

Comment:
Is there any reason why these tests are executed just against REST endpoint? I would expect REST and WS API to behave the same.

Reply:
@sutaakar these are behaviors that currently apply only to the REST service, see _RestSpecManagementServiceImpl_ class, which does some extra validations on top of _SpecManagementServiceImpl_ service. @mswiderski wdyt, should these be moved into base service class?

Reply:
@cristianonicolai yes, I think we need to have that validation moved to base classes to make sure it consistent and work the same way and the only thing that is different is the protocol.

Reply:
@sutaakar @mswiderski sure, updated PR so that both protocols have similar behavior.